### PR TITLE
Exit on failure when building the agent image

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR updates the `install.sh` to exit on failure when building agent images

## Fixes

https://github.com/valory-xyz/open-autonomy/issues/1492

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by 0xArdi — DISMISSED** [review]

**Review by 8ball030 — APPROVED** [review]
thats really nice. I like the test as well.

**Comment by Karrenbelt** [comment]
> thats really nice. I like the test as well.

tests don't like @angrybayblade though :disappointed:

File: tests/test_autonomy/test_cli/test_build_image.py [hunk 1]
```diff
@@ -41,23 +45,32 @@ class TestBuildImage(BaseCliTest):
...
+
+        os.chdir(self.t)
+
+        service_config, *_ = get_dummy_service_config()
+        service_config["agent"] = str(self.package_id.public_id)
+        service_file = self.t / "service.yaml"
+
+        with open_file(service_file, "w+") as fp:
+            service = Service.from_json(service_config)
+            ConfigLoader(Service.schema, Service).dump(service, fp)
```
Comment (angrybayblade) [inline tests/test_autonomy/test_cli/test_build_image.py:73]: Use agent hashes from the latest tag to build the images during the tests.

File: tests/test_autonomy/test_cli/test_build_image.py [hunk 2]
```diff
@@ -115,3 +139,16 @@ def test_service_file_missing(
         assert result.exit_code == 1, result.output
         assert "No such file or directory: " in result.output
         assert "service.yaml" in result.output
+
+    @skip_docker_tests
```
Comment (angrybayblade) [inline tests/test_autonomy/test_cli/test_build_image.py:143]: We don't run docker integration tests on macOS and windows

**Review by 8ball030 — APPROVED** [review]

File: .github/workflows/main_workflow.yml [hunk 3]
```diff
@@ -175,6 +175,7 @@ jobs:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
+          fetch-depth: 0
```
Comment (angrybayblade) [inline .github/workflows/main_workflow.yml:178]: Required for image build test to work

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**